rfdetr/datasets/yolo.py

```diff
@@ -0,0 +1,364 @@
+        square_resize_div_64 = False
+    
+    # Choose appropriate transforms
+    if square_resize_div_64:
+        dataset = YoloDetection(
+            img_folder, 
+            labels_folder,
+            data_yaml_path,
+            transforms=make_coco_transforms_square_div_64(
+                image_set, 
+                resolution, 
+                multi_scale=args.multi_scale, 
+                expanded_scales=args.expanded_scales
+            )
+        )
+    else:
+        dataset = YoloDetection(
+            img_folder, 
+            labels_folder,
+            data_yaml_path,
+            transforms=make_coco_transforms(
+                image_set, 
+                resolution, 
+                multi_scale=args.multi_scale, 
+                expanded_scales=args.expanded_scales
+            )
+        )
+    
+    return dataset 
+
+
+class YoloDetection(torch.utils.data.Dataset):
+    """Dataset for YOLO format annotations"""
+    def __init__(self, img_folder, labels_folder, data_yaml_path, transforms=None):
+        super(YoloDetection, self).__init__()
+        self.img_folder = img_folder
+        self.labels_folder = labels_folder
+        self.transforms = transforms
+        self.img_files = sorted(glob.glob(os.path.join(img_folder, '*.jpg')) + 
+                        glob.glob(os.path.join(img_folder, '*.jpeg')) + 
+                        glob.glob(os.path.join(img_folder, '*.png')))
+        
+        # Read class names from data.yaml
+        with open(data_yaml_path, 'r') as f:
+            data = yaml.safe_load(f)
+            self.class_names = data.get('names', [])
+            
+        print(f"Loaded {len(self.class_names)} classes from YOLO dataset: {self.class_names}")
+            
+        self.class_to_coco_id = {i: i for i in range(len(self.class_names))}
+        
+        # Image IDs start from 1
+        self.ids = [i+1 for i in list(range(len(self.img_files)))]
+        
+        # Cache for the COCO-like API
+        self._coco = None
+        
+    def __len__(self):
+        return len(self.img_files)
+    
+    def __getitem__(self, idx):
+        # Convert idx to integer if it's a string
+        if isinstance(idx, str):
+            idx = int(idx)
+        img_path = self.img_files[idx]
+        image_id = self.ids[idx]
+
+        # Get label file path (same name as image but with .txt extension)
+        base_name = os.path.splitext(os.path.basename(img_path))[0]
+        label_path = os.path.join(self.labels_folder, base_name + '.txt')
+        
+        # Load image
+        img = Image.open(img_path).convert('RGB')
+        w, h = img.size
+        
+        target = {}
+        target["image_id"] = torch.tensor([image_id])
+        target["orig_size"] = torch.as_tensor([int(h), int(w)])
+        target["size"] = torch.as_tensor([int(h), int(w)])
+        
+        boxes = []
+        labels = []
+        
+        # Check if label file exists
+        if os.path.exists(label_path):
```

Comment:
let's create a separate `load_txt` function, place it in `rfdetr/util/files.py`.

Please make sure to include any necessary checks. Make sure util is general. Applicable to load any TXT file. ;)